### PR TITLE
Added cronjob to flush invalidated cache types

### PR DESCRIPTION
**Pull request by nordcomputer**

### CODE REVIEW

**Review by serzhik — APPROVED** [review]

**Review by serzhik — APPROVED** [review]

File: Cron/FlushInvalidatedCacheTypes.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
+<?php
+namespace Pronko\SelectiveCache\Cron;
+
+use Magento\Framework\App\Cache\TypeListInterface;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Pronko\SelectiveCache\Logger\CleanCacheLogger;
+
+/** Class FlushInvalidatedCacheTypes flushes invalidated cache types by cronjob */
+
```
Comment (mcspronko) [inline Cron/FlushInvalidatedCacheTypes.php:9]: Please remove extra line between class description and class name

File: Cron/FlushInvalidatedCacheTypes.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
+<?php
+namespace Pronko\SelectiveCache\Cron;
```
Comment (mcspronko) [inline Cron/FlushInvalidatedCacheTypes.php:2]: Please use strict_types here and other files

File: Cron/FlushInvalidatedCacheTypes.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+/** Class FlushInvalidatedCacheTypes flushes invalidated cache types by cronjob */
+
+class FlushInvalidatedCacheTypes
+{
+
+    /**
+     * @var TypeListInterface
+     */
+    protected $_cacheTypeList;
```
Comment (mcspronko) [inline Cron/FlushInvalidatedCacheTypes.php:16]: Please update to `private $cacheTypeList;`

File: Cron/FlushInvalidatedCacheTypes.php [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+    /**
+     * @var ScopeConfigInterface
+     */
+    protected $scopeConfig;
```
Comment (mcspronko) [inline Cron/FlushInvalidatedCacheTypes.php:21]: Please use `private`

File: Cron/FlushInvalidatedCacheTypes.php [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    
+    /**
+     * Method execute flushes invalidated Cache types
+     *
+     * @return void
+     */
+
+    public function execute()
+    {
+        $cronenabled = $this->getConfig('selectivecache/general/enabled');
```
Comment (mcspronko) [inline Cron/FlushInvalidatedCacheTypes.php:52]: Please use `isFlag()` method

File: Cron/FlushInvalidatedCacheTypes.php [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        if ($cronenabled==true) {
+            foreach ($this->_cacheTypeList->getInvalidated() as $invalidatedType) {
+                $this->_cacheTypeList->cleanType($invalidatedType->getData('id'));
+                $cacheLabels[] = $invalidatedType->getData('cache_type');
+            }
+            
+            if (!empty($cacheLabels)) {
+                $logoutput = implode(", ", $cacheLabels);
+                $this->_logger->info(__("Following cache types were automatically cleared: ").$logoutput);
```
Comment (mcspronko) [inline Cron/FlushInvalidatedCacheTypes.php:61]: Use the phase
__("Cache types cleared automatically: %1", implode(',' $cacheLabels))

File: etc/adminhtml/config.xml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Store:etc/config.xsd">
+    <default>
+        <selectivecache>
+            <general>
```
Comment (mcspronko) [inline etc/adminhtml/config.xml:5]: It is not clear that these settings belong to cron. Consider renaming the section to cron

**Review by mcspronko — CHANGES_REQUESTED** [review]

File: Cron/FlushInvalidatedCacheTypes.php (hunk 5 above)
Comment (nordcomputer) [reply Cron/FlushInvalidatedCacheTypes.php:52]: I have not worked yet with that method - do you have an example?

**Comment by nordcomputer** [comment]
ok, I think, I got all your requested changes - happy Reviewing

**Review by mcspronko — APPROVED** [review]

**Pull request merged**